plugins/amazonq/codewhisperer/jetbrains-community/src/software/aws/toolkits/jetbrains/services/codewhisperer/telemetry/CodeWhispererTelemetryService.kt

```diff
@@ -385,59 +390,60 @@ class CodeWhispererTelemetryService {
         popupShownTime: Duration? = null,
     ) {
-        val detailContexts = recommendationContext.details
-        val decisions = mutableListOf<CodewhispererSuggestionState>()
-
-        detailContexts.forEachIndexed { index, detailContext ->
-            val suggestionState = recordSuggestionState(
-                index,
-                sessionContext.selectedIndex,
-                sessionContext.seen.contains(index),
-                hasUserAccepted,
-                detailContext.isDiscarded,
-                detailContext.recommendation.content().isEmpty()
-            )
-            sendUserDecisionEvent(requestContext, responseContext, detailContext, index, suggestionState, detailContexts.size)
+        CodeWhispererService.getInstance().getAllPaginationSessions().forEach { (jobId, state) ->
+            if (state == null) return@forEach
+            val decisions = mutableListOf<CodewhispererSuggestionState>()
+            val details = state.recommendationContext.details
 
-            decisions.add(suggestionState)
-        }
+            details.forEachIndexed { index, detail ->
+                val suggestionState = recordSuggestionState(detail, hasUserAccepted)
+                sendUserDecisionEvent(state.requestContext, state.responseContext, detail, index, suggestionState, details.size)
 
-        with(aggregateUserDecision(decisions)) {
-            // the order of the following matters
-            // step 1, send out current decision
-            previousUserTriggerDecisionTimestamp = Instant.now()
-
-            val referenceCount = if (hasUserAccepted && detailContexts[sessionContext.selectedIndex].recommendation.hasReferences()) 1 else 0
-            val acceptedContent =
-                if (hasUserAccepted) {
-                    detailContexts[sessionContext.selectedIndex].recommendation.content()
-                } else {
-                    ""
+                decisions.add(suggestionState)
+            }
+            LOG.debug { "jobId: $jobId, userDecisions: [${decisions.joinToString(", ")}]" }
+
+            with(aggregateUserDecision(decisions)) {
+                // the order of the following matters
+                // step 1, send out current decision
+                LOG.debug { "jobId: $jobId, userTriggerDecision: $this" }
+                previousUserTriggerDecisionTimestamp = Instant.now()
+
+                val previews = CodeWhispererService.getInstance().getAllSuggestionsPreviewInfo()
+                val recommendation =
+                    if (hasUserAccepted) {
+                        previews[sessionContext.selectedIndex].detail.recommendation
+                    } else {
+                        Completion.builder().content("").references(emptyList()).build()
+                    }
+                val referenceCount = if (hasUserAccepted && recommendation.hasReferences()) 1 else 0
+                val acceptedContent = if (hasUserAccepted) recommendation.content() else ""
```

Comment:
think the if else here is extra, could be `val acceptedContent = recommendation.content`

Reply:
in rejected cases we want the acceptedContent to be "", or at least this is the existing behavior.

Reply:
i mean you already had this logic here, isn't it
```
val recommendation =
                    if (hasUserAccepted) {
                        previews[sessionContext.selectedIndex].detail.recommendation
                    } else {
                        Completion.builder().content("").references(emptyList()).build()
                    }
```

Reply:
yes, this is correct, I forgot I put a recommendation variable there, let me change this

Reply:
fixed